model/scheduler/StorageAggregator.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoScheduler\model\scheduler;
+
+
+use oat\taoScheduler\model\job\JobInterface;
+use oat\taoScheduler\model\SchedulerException;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * Class StorageAggregator
+ * @package oat\taoScheduler\model\scheduler
+ */
+class StorageAggregator implements SchedulerStorageInterface
+{
+    use ServiceLocatorAwareTrait {
+        setServiceLocator as traitSetServiceLocator;
+    }
+    /**
+     * @var SchedulerStorageInterface
+     */
+    private $dynamicStorage;
+
+    /**
+     * @var SchedulerStorageInterface
+     */
+    private $configStorage;
+
+    /**
+     * StorageAggregator constructor.
+     * @param SchedulerStorageInterface $dynamicStorage
+     * @param SchedulerStorageInterface $configStorage
+     */
+    public function __construct(SchedulerStorageInterface $dynamicStorage, SchedulerStorageInterface $configStorage)
+    {
+        $this->dynamicStorage = $dynamicStorage;
+        $this->configStorage = $configStorage;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function add(JobInterface $job, $permanent = true)
+    {
+        if ($permanent) {
+            return $this->dynamicStorage->add($job);
+        } else {
+            return $this->configStorage->add($job);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function remove(JobInterface $job)
+    {
+        try {
+            $result = $this->dynamicStorage->remove($job);
+        } catch (SchedulerException $e) {
```

Comment:
Exception is needed here to distinguish 2 cases:
It was not removed because it does not exist
It was not removed on the level of underlying persistence (like RDS, kv storage)